turtle_graphics/turtle.js

```diff
@@ -0,0 +1,130 @@
+class Turtle {
+  constructor(initialX, initialY) {
+    this.x = initialX;
+    this.y = initialY;
+    this.path = [[this.x, this.y]];
+    this.direction = "x";
+    this.maxY = -Infinity;
+    this.minY = Infinity;
+    this.maxX = -Infinity;
+    this.minX = Infinity;
+  }
+
+  forward(steps) {
+    for (let i = 0; i < steps; i++) {
+      if (this.direction == "x") {
+        this.x++;
+      } else if (this.direction === "-y") {
+          this.y--;
+      } else if (this.direction === "-x") {
+        this.x--;
+      } else {
+        this.y++;
+      }
+      this.path.push([this.x, this.y]);
+    }
+    return this;
+  }
+
+  right() {
+    switch (this.direction) {
+      case "x": 
+        this.direction = "-y";
+        break;
+      case "y": 
+        this.direction = "x";
+        break;
+      case "-x": 
+        this.direction = "y";
+        break;
+      case "-y": 
+        this.direction = "-x";
+        break;
+      default: 
+        return this;
+      } 
+    return this;
+  }
+
+  left() {
+    switch (this.direction) {
+      case "x": 
+        this.direction = "y";
+        break;
+      case "-y": 
+        this.direction = "x";
+        break;
+      case "-x": 
+        this.direction = "-y";
+        break;
+      case "y": 
+        this.direction = "-x";
+        break;
+      default: 
+        return this;
+    }
+    return this;
+  }
+
+  canvasSize() {
+    for (let position of this.path) {
+      if (position[1] > this.maxY) {
+        this.maxY = position[1];
+      }
+    }
+    for (let position of this.path) {
+      if (position[1] < this.minY) {
+        this.minY = position[1];
+      }
+    }
+    for (let position of this.path) {
+      if (position[0] > this.maxX) {
+        this.maxX = position[0];
+      }
+    }
+    for (let position of this.path) {
+      if (position[0] < this.minX) {
+        this.minX = position[0];
+      }
+    }
+    return (this);
+  }
+  print() {
+    const full = "\u2588";
+    const empty = " ";
+    this.canvasSize();
```

Comment:
I would expect a method called `canvasSize` to _return_ a size. In this case it's more of an `setCanvasBoundaries` or `updateCanvasBoundaries` or something like that